odilo.py

```diff
@@ -0,0 +1,598 @@
+
+        return HTTP.post_with_timeout(url, payload, headers=headers, **kwargs)
+
+
+class MockOdiloAPI(OdiloAPI):
+    @classmethod
+    def mock_collection(cls, _db):
+        library = DatabaseTest.make_default_library(_db)
+        collection, ignore = get_one_or_create(
+            _db, Collection,
+            name="Test Odilo Collection",
+            create_method_kwargs=dict(
+                external_account_id=u'library_id_123',
+            )
+        )
+        integration = collection.create_external_integration(
+            protocol=ExternalIntegration.ODILO
+        )
+        integration.username = u'username'
+        integration.password = u'password'
+        integration.setting(OdiloAPI.LIBRARY_API_BASE_URL).value = u'http://localhost:8080/api/v2'
+        library.collections.append(collection)
+
+        return collection
+
+    def __init__(self, _db, collection, *args, **kwargs):
+        self.access_token_requests = []
+        self.requests = []
+        self.responses = []
+
+        self.access_token_response = self.mock_access_token_response('token')
+        super(MockOdiloAPI, self).__init__(_db, collection, *args, **kwargs)
+
+    def token_post(self, url, payload, headers={}, **kwargs):
+        """Mock the request for an OAuth token.
+        """
+
+        self.access_token_requests.append((url, payload, headers, kwargs))
+        response = self.access_token_response
+        return HTTP._process_response(url, response, **kwargs)
+
+    def mock_access_token_response(self, credential):
+        token = dict(token=credential, expiresIn=3600)
+        return MockRequestsResponse(200, {}, json.dumps(token))
+
+    def queue_response(self, status_code, headers={}, content=None):
+        self.responses.insert(
+            0, MockRequestsResponse(status_code, headers, content)
+        )
+
+    def _do_get(self, url, *args, **kwargs):
+        """Simulate Representation.simple_http_get."""
+        response = self._make_request(url, *args, **kwargs)
+        return response.status_code, response.headers, response.content
+
+    def _do_post(self, url, *args, **kwargs):
+        return self._make_request(url, *args, **kwargs)
+
+    def _make_request(self, url, *args, **kwargs):
+        response = self.responses.pop()
+        self.requests.append((url, args, kwargs))
+        return HTTP._process_response(
+            url, response, kwargs.get('allowed_response_codes'),
+            kwargs.get('disallowed_response_codes')
+        )
+
+
+class OdiloRepresentationExtractor(object):
+    """Extract useful information from Odilo's JSON representations."""
+
+    log = logging.getLogger("OdiloRepresentationExtractor")
+
+    format_data_for_odilo_format = {
+        "ACSM": (
+            Representation.PDF_MEDIA_TYPE, DeliveryMechanism.ADOBE_DRM
+        ),
+        "EBOOK_STREAMING": (
+            Representation.TEXT_HTML_MEDIA_TYPE, DeliveryMechanism.STREAMING_TEXT_CONTENT_TYPE
+        ),
```

Comment:
Rather than defining a new constant here and mapping both 'PDF' and 'EPUB' to that constant so the lookup works, I think it would be simpler to map both 'PDF' and 'EPUB' to identical values in this dictionary.

Reply:
And what about 'delivery_mechanism_to_internal_format' which is just the inverted map?? There will be two identical keys and the second will overwrite the first one.